### ISSUE TITLE
As written, only one user can deposit to the sidechain 

### ISSUE THREAD

**Issue by rot13maxi** [supertestnet/handcrank#1]
In this step:
> She asks those 51 miners to sign a multisig covenant and then delete their keys

The miners delete the private keys that correspond to the public keys that they posted in the coinbase tx. That means that once a user has gotten a miner to participate in creating the covenant, the miner no longer has a private key that they can use to cooperate. I think you might consider either having the miner sign with a tweaked key or some derived key. Maybe miners post an xpub and part of the covenant payload is the derivation path of the key that was used in _that_ covenant. 
There might also be something clever with adaptors.

**Comment by supertestnet** [comment]
That is true. I don't like the idea of commiting to an xpub because that makes it hard to delete their keys. You can't delete "just one key" from an xprv, you have to either delete the whole thing (making it useless for the future) or keep the whole thing, including all of the keys derived from it.

Perhaps instead miners could regularly put a message on the sidechain containing a list of 1000 or so pubkeys, all with independent private keys, and those are the keys users must pick from in order to peg in. That would also eliminate the need to put any sidechain messages into bitcoin's blockchain.

**Comment by supertestnet** [comment]
> You can't delete "just one key" from an xprv, you have to either delete the whole thing (making it useless for the future) or keep the whole thing, including all of the keys derived from it.

Nevermind, I thought of a way to do this. Every xprv/xpub keypair "contains" (so to speak) a practically infinite number of "regular" keypairs, and each of those keypairs can be converted into another xprv/xpub pair with another practically infinite number of keypairs inside it. You can use this property in a commitment scheme to reduce the number of keys you have to commit to up front.

Miners can create a single xprv/xpub pair -- which I will call x1 -- and commit its xpub on the sidechain or on bitcoin. The handcrank protocol can define that the first user who wants to peg in must use the pubkey of the --first-- keypair derived from x1 as part of their multisig covenant. When the covenant is complete the miner must save the private key of the --second-- keypair derived from x1, convert that private key into a new xprv/xpub pair which I will call x2, and delete the private key to x1. Now they treat x2 like x1 for the next peg in and loop.

Users can follow along without needing the miner to explicitly reference their new key because users can detect that the first pubkey of x1 was used and they can independently derive the pubkey of x2 from x1. I think this allows miners to commit a single key, use it forever, and yet still retain the ability to delete the private keys of "used" keypairs. You just define that every subsequent user must detect which pubkeys  have been used and peg in using the next available pubkey in the trail of xpubs.

Perhaps people more knowledgeable than me in the bip32 field can correct me if I've neglected something. I've got a nagging feeling that if you have a "child" xprv and a "parent" xpub you can derive the "parent" xprv. But there might be an easy solution to that, such as using hardened derivation.
